rules/import-style.js

```diff
@@ -0,0 +1,320 @@
+			checkDynamicImport = true,
+			checkRequire = true
+		} = {}
+	] = context.options;
+
+	styles = extendDefaultStyles ?
+		defaultsDeep({}, styles, defaultStyles) :
+		styles;
+
+	styles = new Map(
+		Object.entries(styles).map(
+			([moduleName, styles]) =>
+				[moduleName, new Map(Object.entries(styles))]
+		)
+	);
+
+	const report = (node, moduleName, actualImportStyles, allowedImportStyles, isRequire = false) => {
+		if (!allowedImportStyles || allowedImportStyles.size === 0) {
+			return;
+		}
+
+		let effectiveAllowedImportStyles = allowedImportStyles;
+
+		if (isRequire) {
+			// For `require`, `'default'` style allows both `x = require('x')` (`'namespace'` style) and
+			// `{default: x} = require('x')` (`'default'` style) since we don't know in advance
+			// whether `'x'` is a compiled ES6 module (with `default` key) or a CommonJS module and `require`
+			// does not provide any automatic interop for this, so the user may have to use either of theese.
+			if (allowedImportStyles.get('default') && !allowedImportStyles.get('namespace')) {
+				effectiveAllowedImportStyles = new Map(allowedImportStyles);
+				effectiveAllowedImportStyles.set('namespace', true);
+			}
+		}
+
+		if (actualImportStyles.every(style => effectiveAllowedImportStyles.get(style))) {
+			return;
+		}
+
+		const data = {
+			allowedStyles: joinOr([...allowedImportStyles.keys()]),
+			moduleName
+		};
+
+		context.report({
+			node,
+			messageId: MESSAGE_ID,
+			data
+		});
+	};
+
+	return templates.visitor({
+		ImportDeclaration(node) {
+			if (!checkImport) {
+				return;
+			}
+
+			const moduleName = getStringIfConstant(node.source, context.getScope());
+
+			const allowedImportStyles = styles.get(moduleName);
+			const actualImportStyles = getActualImportDeclarationStyles(node);
+
+			report(node, moduleName, actualImportStyles, allowedImportStyles);
+		},
+
+		'ExpressionStatement > ImportExpression'(node) {
+			if (!checkDynamicImport) {
+				return;
+			}
+
+			const moduleName = getStringIfConstant(node.source, context.getScope());
+			const allowedImportStyles = styles.get(moduleName);
+			const actualImportStyles = ['unassigned'];
+
+			report(node, moduleName, actualImportStyles, allowedImportStyles);
+		},
+
+		[assignedDynamicImportTemplate](node) {
+			if (!checkDynamicImport) {
+				return;
+			}
+
+			const assignmentTargetNode = assignedDynamicImportTemplate.context.getMatch(assignmentTargetVariable);
+			const moduleNameNode = assignedDynamicImportTemplate.context.getMatch(moduleNameVariable);
+			const moduleName = getStringIfConstant(moduleNameNode, context.getScope());
+
+			if (!moduleName) {
+				return;
+			}
+
+			const allowedImportStyles = styles.get(moduleName);
+			const actualImportStyles = getActualAssignmentTargetImportStyles(assignmentTargetNode);
+
+			report(node, moduleName, actualImportStyles, allowedImportStyles);
+		},
+
+		'ExpressionStatement > CallExpression[callee.name=\'require\']'(node) {
+			if (!checkRequire) {
```

Comment:
`ExpressionStatement > CallExpression[callee.name="require"][arguments.length=1][arguments.0.type="Literal"]`

This work?